automation/features/pageobjects/page.js

```diff
@@ -42,20 +42,59 @@ class Page {
     return $("[data-test-id='connect-to-eth-wallet-button']");
   }
   get disconnectWalletBtn() {
-    return $$(".button-link=Disconnect");
-  } //TODO add test id
+    return browser.getByTestId("disconnect-vega-wallet-btn-link");
+  }
   get connectMetaMask() {
     return $(".web3modal-provider-name*=Injected");
   }
 
   goTo(path) {
     browser.url(path);
+    browser.waitUntil(
+      function () {
+        const state = browser.execute(function () {
+          return document.readyState;
+        });
+        return state === "complete";
+      },
+      {
+        timeout: 10000,
+        timeoutMsg: "Page did not fully load in 10 seconds",
+      }
+    );
   }
 
   connectToEthWallet() {
     this.connectEthWallet.click();
     this.connectMetaMask.click();
   }
+
+  confirmTransactionSpeedUp() {
+    browser.waitUntil(() => browser.getWindowHandles().length > 1);
+    browser.switchWindow("MetaMask Notification");
```

Comment:
I'd like to make a class for meta mask interactions. I understand this is on the base page class, which conceptualy is elements always on the page under test. So would be cool to have:


```
class MM {
   setup () {/* Do things required to import network ect. */}
   confirm () {/* Do things required to import network ect. */}
   viewAssetList () {/* Do things required to import network ect. */}
}
```

view asset list will be needed for some stuff relating to adding tokens to wallet later. 

Can do this as a follow up if you don't want to do it in this PR, else create an issue and we can do soon.

Reply:
we can do this as a follow up would be better thanks

Reply:
https://github.com/vegaprotocol/token-frontend/issues/1281